File: paper 3/Fig1.m
% 計算給定 Cpk 值的 PPM 界限
function [lower_ppm, upper_ppm] = calculate_ppm_bounds(cpk)
% 單邊規格計算
 upper_ppm = (1 - normcdf(3 * cpk)) * 1e6;
% 雙邊規格計算（製程中心對稱情況）
 lower_ppm = 2 * (1 - normcdf(3 * cpk)) * 1e6;
end

cpk_seq = 0.6:0.01:2.0;

% 計算每個 Cpk 值的 PPM 界限
results = zeros(length(cpk_seq), 2);
for i = 1:length(cpk_seq)
 [results(i,1), results(i,2)] = calculate_ppm_bounds(cpk_seq(i));
end

% 繪製圖形
figure;
plot(cpk_seq, results(:,1), 'LineWidth', 1);
hold on;
plot(cpk_seq, results(:,2), 'LineWidth', 1);

% 設定座標軸標籤和範圍
xlabel('C_{pk}', 'FontSize', 12);
ylabel('PPM', 'FontSize', 12);
ylim([0 80000]);
xlim([0.6 2.0]);

% 設定 y 軸刻度
yticks(0:20000:80000);

% 調整圖形邊界
set(gca, 'Position', [0.15 0.2 0.8 0.7]);
hold off;

File: paper 3/Fig 1.R
# 定義函數計算給定 Cpk 值的 PPM 上下界
calculate_ppm_bounds <- function(cpk) {
  # 單邊規格的情況: 使用常態分配計算超出規格的 PPM 值
  upper_ppm <- (1 - pnorm(3 * cpk)) * 1e6
  
  # 雙邊規格的情況: 當製程置中時(對稱情況)的下界
  lower_ppm <- 2 * (1 - pnorm(3 * cpk)) * 1e6
  
  return(c(lower_ppm, upper_ppm))
}

# Cpk 從 0.6 到 2.0,間隔 0.01
cpk_seq <- seq(0.6, 2.0, by = 0.01)

# 對每個 Cpk 值計算其 PPM 上下界
results <- t(sapply(cpk_seq, calculate_ppm_bounds))

par(mar = c(5, 5, 2, 2))  # 調整邊界大小

# 建立主要圖形
plot(cpk_seq, results[,1], type = "l", 
     xlab = expression(C[pk]),    
     ylab = "PPM",              
     ylim = c(0, 80000),         
     xlim = c(0.6, 2.0),         
     yaxt = "n",                 
     cex.lab = 1.2,             # 調整標籤大小
     lwd = 1)                   # 設定線條寬度

# 加入上界線
lines(cpk_seq, results[,2], lwd = 1)

# 建立自訂的 Y 軸刻度
axis(2, at = seq(0, 80000, by = 20000), las = 1)

# 在圖形下方加入標題
title(sub = expression(paste("Fig. 1. The bounds on nonconforming units in PPM versus ", C[pk], ".")),
      line = 3.5)